js/parts/Dynamics.js

```diff
@@ -1103,7 +1103,10 @@ extend(Series.prototype, /** @lends Series.prototype */ {
 
             // Do the merge, with some forced options
             options = merge(oldOptions, animation, {
-                index: series.index,
+                // When oldOptions.index is null it should't be cleared.
+                // Otherwise navigator series will have wrong indexes (#10193).
+                index: oldOptions.index === undefined ?
+                    series.index : oldOptions.index,
```

Comment:
Use the shorter `pick(oldOptions.index, series.index)`

Reply:
I haven't used `pick` because it return the first value that is not `null` or `undefined`. Here I want only `undefined` to be true.

Reply:
Oh yes, it's not the same! Sorry.